### DISCUSSION TITLE
Machine translation of missing keys

### DISCUSSION THREAD

**Discussion by andrey-helldar** [Laravel-Lang/common#41] [Q&A]
@caouecs, sorry for DDoS with PR :)

I thought it would be correct to create a separate PR for each localization. This will get rid of the noise of other people's files.

PS: to translate phrases, I relied on a script that receives a list of untranslated keys for the status.md page. Therefore, some nested keys, for example, "validation.php", were not translated.

Nevertheless, in my opinion, this is much better than nothing.

During processing, I requested the translation of only those keys that were not translated. I did not touch the keys that people translated.

Another problem I ran into was unsupported localization.

When requesting a translation, I pass the localization code in the `ISO 639-1` format ([service requirement](https://cloud.yandex.com/en-ru/docs/translate/api-ref/Translation/translate)), which does not contain some of the values. For example, `fil`. I took the names of localizations from ours up to the `_` symbol.

I couldn't use Google Translate as they charge $40 per hour to access their API. The Yandex.Translate service completed the translation of all phrases in about 1.5 hours and charged $5.28.

**Before:**

![image](https://user-images.githubusercontent.com/10347617/115113588-cc358300-9f93-11eb-8466-c7c726796371.png)

**After:**

![image](https://user-images.githubusercontent.com/10347617/115113742-70b7c500-9f94-11eb-9c9d-d6fe37ceb3d1.png)

By the way, I also noticed that some words, for example, the names of countries, may coincide during translation - now the script does not fill in the exclusion files, so they are shown as untranslated.
I tried to run a script to fill in such fields, but I ran into a problem that really untranslated phrases also get into the exclusion files. Therefore, I did not fill in the exclusion files.

**Comment by caouecs** [comment]
Good, but I prefer to contact each referents or last people who worked on a language, just to have fixes ( if needed )

**Reply by andrey-helldar** [reply]
I totally agree with you ☜(ﾟヮﾟ☜)

**Reply by andrey-helldar** [reply]
We have referents in some languages. What are we going to do with the rest of the PRs?

**Reply by caouecs** [reply]
We can contact the last people who worked on a language

**Reply by andrey-helldar** [reply]
In principle, we can.

For example:

![image](https://user-images.githubusercontent.com/10347617/115125981-76cd9600-9fd4-11eb-8277-d1a2df55e888.png)

![image](https://user-images.githubusercontent.com/10347617/115125985-7cc37700-9fd4-11eb-86af-f9763f805f5c.png)

**Reply by andrey-helldar** [reply]
But not everywhere)

![image](https://user-images.githubusercontent.com/10347617/115126038-eba0d000-9fd4-11eb-9e71-b1be7d8d7559.png)

**Reply by caouecs** [reply]
Yes, because you have update the directory ;)

**Reply by andrey-helldar** [reply]
@caouecs, Hey! Sorry for not answering. At work, we are migrating the server and applications. I will be very busy for the next couple of days. As soon as I am free, I will do everything necessary according to my PR.

**Reply by caouecs** [reply]
No problem, I ask help to last translators, and the result is good ;)

**Comment by caouecs** [comment]
Translation of localizations is completed by 40%

**Reply by caouecs** [reply]
42%

**Reply by caouecs** [reply]
But I think I spam a lot of people 😇

**Reply by caouecs** [reply]
45%

**Reply by andrey-helldar** [reply]
It looks like spam from you, yes. But from the user's side these are 1-2 letters)

However, as far as I could see, until we started asking for help in translation, the developers were not interested in our project. But as soon as we contacted them personally, they immediately agreed to help.

¯\_(ツ)_/¯

**Reply by caouecs** [reply]
I think they didn't know that we have a lot of updates on files

**Reply by andrey-helldar** [reply]
I think so too. It could also be that they stopped using our package.

**Reply by caouecs** [reply]
52%

**Comment by caouecs** [comment]
Yandex doesn't love Qatar ;)

**Reply by andrey-helldar** [reply]
I'll tell you more - he loves few people 😂

**Comment by caouecs** [comment]
We need to make a calendar with translations not reviewed, and a strategy to explain the reasons why the translations are not very very good.

**Reply by andrey-helldar** [reply]
Hmm...

![image](https://user-images.githubusercontent.com/10347617/115926228-2bf7c680-a48b-11eb-8df1-827ee49cdb72.png)

**Reply by andrey-helldar** [reply]
See https://github.com/Laravel-Lang/lang/pull/1713

Personally, I don't like the look of the emoji table.

But I have not yet figured out how to make it beautifully.

**Reply by caouecs** [reply]
![Capture d’écran 2021-04-24 à 10 31 43](https://user-images.githubusercontent.com/447849/115952811-7e1b0500-a4e8-11eb-93e7-133a32328717.png)

https://laravel-lang.github.io/lang/status.html

I don't see all table on Safari and Firefox

**Reply by andrey-helldar** [reply]
![image](https://user-images.githubusercontent.com/10347617/116054460-5cef1b80-a684-11eb-83da-2b43c3638d3b.png)

Google Chrome too

Google Pages has a limited content width. Due to the addition of new characters, the width has become larger. I will reduce the number of displayed columns from 8 to 6.

**Reply by andrey-helldar** [reply]
https://github.com/Laravel-Lang/lang/pull/1724

**Comment by caouecs** [comment]
I merge all without reviewer, and I indicate that it's possible to update translations after

**Comment by caouecs** [comment]
88% after accepted all machine translations and do some fixes

**Comment by dansleboby** [comment]
You should deepl.com free 500k and ther translation are way better!

**Reply by caouecs** [reply]
Yes, we know. When I find time, I will do a test ;)
